huaweicloud/services/er/data_source_huaweicloud_er_instances.go

```diff
@@ -0,0 +1,251 @@
+							Computed:    true,
+							Description: `The instance ID.`,
+						},
+						"asn": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: `The BGP AS number of the ER instance.`,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of the instance.`,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The description of the instance.`,
+						},
+						"status": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The current status of the instance.`,
+						},
+						"enterprise_project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of enterprise project to which the instance belongs.`,
+						},
+						"tags": {
+							Type:        schema.TypeMap,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: `The key/value pairs to associate with the instance.`,
+						},
+						"created_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The creation time of the instance.`,
+						},
+						"updated_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The last update time of the instance.`,
+						},
+						"enable_default_propagation": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: `Whether to enable the propagation of the default route table.`,
+						},
+						"enable_default_association": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: `Whether to enable the association of the default route table.`,
+						},
+						"auto_accept_shared_attachments": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: `Whether to automatically accept the creation of shared attachment.`,
+						},
+						"default_propagation_route_table_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the default propagation route table.`,
+						},
+						"default_association_route_table_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the default association route table.`,
+						},
+						"availability_zones": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: `The availability zone list where the ER instance is located.`,
+						},
+					},
+				},
+				Description: `All instances that match the filter parameters.`,
+			},
+		},
+	}
+}
+
+func filterInstancesByTags(d *schema.ResourceData, all []instances.Instance) ([]instances.Instance, error) {
+	filter := map[string]interface{}{
+		"Name": d.Get("name"),
+	}
```

Comment:
The name is optional, maybe use `d.getOK("name")` first, to check if it is exist.